### ISSUE TITLE
bug: color is missing if I print a diff with rich.syntax

### ISSUE THREAD

**Issue by 15r10nk** [pawamoy/markdown-exec#43]
### Description of the bug

I use markdown-exec to include the output of pytest in my documentation.
I use rich to generate some diffs during this run and have the problem that the red and green colors are missing in the diff.

![image](https://github.com/pawamoy/markdown-exec/assets/44680962/8f78105e-4b01-4a18-95f2-65847c86738d)

I was able to reproduce this bug without pytest, but I have no idea if this is a markdown-exec or rich problem.

Can you please look into?

### To Reproduce
see the [repository](https://github.com/15r10nk/bug_markdown_exec_rich_syntax/tree/main)

```
pip install -r requirements.txt
mkdocs serve
```

### Expected behavior

I would expect that markdown-exec is able to produce the same colors which I also see in the terminal.

### Environment information

```bash
python -m markdown_exec.debug  # | xclip -selection clipboard
```

- __System__: Linux-6.1.0-18-amd64-x86_64-with-glibc2.36
- __Python__: cpython 3.12.1
- __Environment variables__:
- __Installed packages__:
  - `markdown-exec` v1.8.0

**Comment by pawamoy** [comment]
Hey @15r10nk, thanks for the report!

Interesting. My first intuition is that Rich uses ANSI codes that are not supported by pygments-ansi. I'll replicate the issue locally and report back :slightly_smiling_face:

**Comment by pawamoy** [comment]
Here's the ANSI output of Rich:

```
\e[31m red\e[0m
\e[32m green\e[0m
\e[48;5;235m                                                                   \e[0m
\e[38;5;231;48;5;235m \e[0m\e[38;5;231;48;5;235m code\e[0m\e[48;5;235m                                                             \e[0m
\e[38;5;204;48;5;235m- a\e[0m\e[48;5;235m                                                                \e[0m
\e[38;5;148;48;5;235m+ b\e[0m\e[48;5;235m                                                                \e[0m
\e[48;5;235m                                                                   \e[0m
```

I've manually escaped the escape sequence (`\e`) and removed some padding.

**Comment by pawamoy** [comment]
Here it is with the standard color mode:

```
\e[31m red\e[0m
\e[32m green\e[0m
\e[40m                                      \e[0m
\e[97;40m \e[0m\e[97;40m code\e[0m\e[40m                                \e[0m
\e[91;40m- a\e[0m\e[40m                                   \e[0m
\e[92;40m+ b\e[0m\e[40m                                   \e[0m
\e[40m                                      \e[0m
```

**Comment by pawamoy** [comment]
It's a bit better with this:

```python
console=Console(color_system="standard")

console.print("[red] red")
console.print("[green] green")


console.print(
    Syntax(
        """
  code
- a
+ b
        """,
        "diff",
        theme="ansi_light",
        background_color="#ffffff",
    ),
)
```

**Comment by pawamoy** [comment]
Aha! This translates as a `-Color-BrightRed` CSS class but there's no such class in the style sheet, only `-Color-Bold-Red`. Let me try to update the style sheet :smile:

**Comment by pawamoy** [comment]
Should be fixed in v1.8.1. Try with the standard color system and the ansi_light/ansi_dark theme and let me know if there are any other issues :slightly_smiling_face:

**Comment by 15r10nk** [comment]
Thank you, it works for me now.
